metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py

```diff
@@ -272,10 +275,9 @@ def _extract_lineage(
         )
 
         # Write the lineage to the datajob object.
-        datajob.inlets.extend(DatasetUrn.create_from_string(urn) for urn in input_urns)
-        datajob.outlets.extend(
-            DatasetUrn.create_from_string(urn) for urn in output_urns
-        )
+        datajob.inlets.extend(entities_to_dataset_urn_list(input_urns))
```

Comment:
how about a `entities_to_urn_list(list, allowed_entities={"dataset, "datajob"})`

Reply:
If we use common method to extend both iolets and upstrem_jobs, we get lint error as:
`src/datahub_airflow_plugin/datahub_listener.py:279: error: Argument 1 to "extend" of "list" has incompatible type "list[DatasetUrn | DataJobUrn]"; expected "Iterable[DatasetUrn]"  [arg-type]`
with below method definition:
`def entities_to_urn_list(
    iolets: List[str], allowed_entities: Set[str]
) -> List[Union[DatasetUrn, DataJobUrn]]:
    urn_list: List[Union[DatasetUrn, DataJobUrn]] = []
    for let in iolets:
        entity = guess_entity_type(let)
        if entity in allowed_entities:
            if entity == "dataset":
                urn_list.append(DatasetUrn.create_from_string(let))
            elif entity == "dataJob":
                urn_list.append(DataJobUrn.create_from_string(let))
    return urn_list`